framework/mesh/mesh_continuum/mesh_continuum.cc

```diff
@@ -381,74 +381,80 @@ MeshContinuum::CheckPointInsideCell(const Cell& cell, const Vector3& point) cons
 {
   const auto& grid_ref = *this;
 
+  // Check each cell edge. A point inside the cell will return a negative value. A point on either
+  // edge will return a zero value, and a point outside the cell will return a positive value.
   if (cell.Type() == CellType::SLAB)
   {
     const auto& v0 = grid_ref.vertices[cell.vertex_ids[0]];
     const auto& v1 = grid_ref.vertices[cell.vertex_ids[1]];
-
-    // Check each cell edge. A point inside the cell will return a negative value. A point on either
-    // edge will return a zero value, and a point outside the cell will return a positive value.
-    if (((v0.z - point.z) * (v1.z - point.z)) > 0.0)
-      return false;
+    return ((v0.z - point.z) * (v1.z - point.z)) <= 0.0;
   }
-  else if (cell.Type() == CellType::POLYGON)
+
+  // Check each face. A point inside the face will give a negative value, a point on the face
```

Comment:
This is a significant change. If the polyhedra are convex (they should be, yeah?), this is valid.